src/common.rs

```diff
@@ -199,3 +217,32 @@ pub fn create_frontend_config_file(keycloak_url: &str) -> Result<()> {
 
     Ok(())
 }
+
+pub async fn check_and_store_ods_version(ods_socket_path: &str) -> Result<()> {
+    let status_response = get_status(ods_socket_path)
+        .await
+        .context("Failed to get status from socket client")?;
+
+    let Some(omnect_device_service_version) =
+        &status_response.system_info.omnect_device_service_version
+    else {
+        bail!("failed to get omnect_device_service_version from status response")
+    };
+
+    let ods_version = omnect_device_service_version.clone();
+
+    // compare to MIN_ODS_VERSION
+    let min_version = Version::parse(MIN_ODS_VERSION).expect("parse MIN_ODS_VERSION");
+    let current_version = Version::parse(&ods_version).expect("parse ods_version");
+    let is_below_min = current_version < min_version;
+    {
+        let mut version_check = VERSION_CHECK.lock().unwrap();
+        *version_check = Some(VersionCheckResult {
+            min_version: MIN_ODS_VERSION.to_string(),
+            current_version: ods_version.clone(),
+            is_below_min,
+        });
+    }
+
+    Ok(())
+}
```

Comment:
1. we should get rid of `expect` and `unwrap`
2. `OnceLock` would be better as `LazyLock`. You could move the code into `get_or_init` function

Reply:
refactored to used OnceLock and removed unnecessary Mutex handling.